### PR TITLE
Don't overwrite with zeroes after unlinking the file

### PR DESCRIPTION
**Pull request by jakubkaczor**
* **What does this PR do?**
Removes overwriting file contents with zeroes in the function for unlinking
files.

* **Screenshots (if relevant)**

* **What are the relevant issue numbers?**
neomutt/neomutt#3219

### CODE REVIEW

**Comment by gahr** [comment]
Opening/closing the file can die too, I guess.

**Comment by xHire** [comment]
BTW, shouldn’t the function also call `close(fd);` at the end?

**Comment by jakubkaczor** [comment]
@xHire I don't think this commit changes the behavior within this regard, does it? I know next to nothing about the programming language C, but I felt this task was simple enough I could do it. 😅 Feel free to argue/edit.

**Comment by xHire** [comment]
@jakubkaczor You’re right, your commit doesn’t change this behaviour. But I thought it would be worth pointing it out while this function is being changed after all those years. :·)

Your changes look good to me.

**Review by gahr — APPROVED** [review]

**Comment by gahr** [comment]
@jakubkaczor we noticed that we're leaking the `fd` now. You should `close(fd)` it on all paths.

File: mutt/file.c [hunk 1]
```diff
@@ -216,17 +216,10 @@ void mutt_file_unlink(const char *s)
...
   {
-    unlink(s);
-    char buf[2048] = { 0 };
-    while (st.st_size > 0)
-    {
-      fwrite(buf, 1, MIN(sizeof(buf), st.st_size), fp);
-      st.st_size -= MIN(sizeof(buf), st.st_size);
-    }
-    mutt_file_fclose(&fp);
+    mutt_file_fclose(&fd);
```
Comment (gahr) [inline mutt/file.c:222]: `mutt_file_fclose` takes a `FILE**`, you have an `int fd` :) you can just `close(fd)` it.

**Comment by jakubkaczor** [comment]
> @jakubkaczor we noticed that we're leaking the `fd` now. You should `close(fd)` it on all paths.

I've made the changes. Is it ok, now?

File: mutt/file.c (hunk 1 above)
Comment (jakubkaczor) [reply mutt/file.c]: Right. Corrected. 😊

**Comment by gahr** [comment]
> > @jakubkaczor we noticed that we're leaking the `fd` now. You should `close(fd)` it on all paths.
> 
> @gahr, I've made the changes. Is it ok, now?

You don't want a condition on `if (fd)`: the file descriptor `fd` is a non-negative integer. Whatever it is, it needs to be closed :)

**Review by gahr — APPROVED** [review]
Thanks a lot!

**Comment by jakubkaczor** [comment]
Thank you! 😊

**Pull request merged**